playback-sdk-android/src/main/java/com/streamamg/player/ui/BackgroundPlaybackService.kt

```diff
@@ -0,0 +1,111 @@
+private const val NOTIFICATION_ID = 1
+private const val EMPTY_CHANNEL_DESCRIPTION = 0
+
+class BackgroundPlaybackService : Service() {
+    // Binder given to clients
+    private val binder = BackgroundBinder()
+    private var bound = 0
+
+    private var player: Player? = null
+    private lateinit var playerNotificationManager: PlayerNotificationManager
+
+    private val customActionReceiver = object : CustomActionReceiver {
+        override fun createCustomActions(context: Context): Map<String, NotificationCompat.Action> =
+            emptyMap()
+
+        override fun getCustomActions(player: Player) = if (!player.isPlaying && bound == 0) {
+            listOf(PlayerNotificationManager.ACTION_STOP)
+        } else {
+            emptyList()
+        }
+
+        override fun onCustomAction(player: Player, action: String, intent: Intent) {
+            if (action == PlayerNotificationManager.ACTION_STOP) stopSelf()
+        }
+    }
+
+    /**
+     * Class used for the client Binder. Because we know this service always
+     * runs in the same process as its clients, we don't need to deal with IPC.
+     */
+    inner class BackgroundBinder : Binder() {
+        // Return this instance of Player so clients can use the player instance
+        val player get() = this@BackgroundPlaybackService.player
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        val playerConfig = PlayerConfig(key = PlaybackSDKManager.bitmovinLicense)
+        player = Player(this, playerConfig)
+
+        // Create a PlayerNotificationManager with the static create method
+        // By passing null for the mediaDescriptionAdapter, a DefaultMediaDescriptionAdapter will be used internally.
+        playerNotificationManager = PlayerNotificationManager.createWithNotificationChannel(
+            this,
+            NOTIFICATION_CHANNEL_ID,
+            R.string.control_notification_channel,
+            EMPTY_CHANNEL_DESCRIPTION,
+            NOTIFICATION_ID,
+            DefaultMediaDescriptor(assets),
+            customActionReceiver
+        ).apply {
+            setNotificationListener(object : NotificationListener {
+                override fun onNotificationStarted(
+                    notificationId: Int,
+                    notification: Notification
+                ) {
+                    startForeground(notificationId, notification)
+                }
+
+                override fun onNotificationCancelled(notificationId: Int) {
+                    stopSelf()
+                }
+            })
+
+            // Attaching the Player to the PlayerNotificationManager
+            setPlayer(player)
+        }
+    }
+
+    override fun onDestroy() {
+        playerNotificationManager.setPlayer(null)
+        player?.destroy()
+        player = null
+
+        stopSelf()
+
+        super.onDestroy()
+    }
+
+    override fun onBind(intent: Intent): IBinder {
+        bound++
```

Comment:
if (bound == 0 && player?.isPlaying == false) {
 stopSelf()
}
Should we make this check and stop when we unbind?

Reply:
@zachtom I followed what bitmovin suggested for unbind the service [here](https://github.com/bitmovin/bitmovin-player-android-samples/blob/6c6125fe70e7845f933be63abfcb286236ea8f77/BackgroundPlaybackKotlin/src/main/java/com/bitmovin/player/samples/playback/background/BackgroundPlaybackService.kt#L104).
Anyway if you think it would be better to add that check I'll try to implement and test it.

Reply:
I marked this as a caution. But if this is the official implementation then thats fine.

Reply:
Done